File: game/tv_show/initial.py
from phone_events import PhoneEvents
from tv_show.press_5 import Press5
from tv_show.tv_show_resources import TvShowResources
from tv_show.video_state import VideoState


class Initial(VideoState):
    def __init__(self, context):
        super().__init__(context, TvShowResources.videos_initial[context.country])

    def process_events(self, phone_events: PhoneEvents):
        super().process_events(phone_events)

        from tv_show.attract import Attract

        if phone_events.hungup:
            return Attract

        if self.video_ended():
            return Press5

        return None

File: game-web/src/tvShow/initial.ts
/** Port of game/tv_show/initial.py */

import { type StateClass } from '../state';
import { TvShowResources } from './tvShowResources';
import { VideoState } from './videoState';
import { Press5 } from './press5';
import type { GameData } from '../gameData';
import type { PhoneEvents } from '../phoneEvents';

// Deferred to avoid circular reference with Attract
import { Attract } from './attract';

export class Initial extends VideoState {
  constructor(context: GameData) {
    super(
      context,
      TvShowResources.videosInitial[context.country],
      false,
      TvShowResources.audioInitial[context.country],
    );
  }

  processEvents(events: PhoneEvents): StateClass | null {
    super.processEvents(events);

    if (events.hungup)     return Attract;
    if (this.videoEnded()) return Press5;

    return null;
  }
}
